RosettaCore/src/main/java/com/hubspot/rosetta/RosettaBinder.java

```diff
@@ -15,6 +16,24 @@ public interface Callback {
     void bind(String key, Object value);
   }
 
+  public void bindList(ArrayNode node, String field, Consumer<Object> valueConsumer) {
+    for (JsonNode value : node) {
+      if (!field.isEmpty()) {
+        if (!value.has(field)) {
+          throw new IllegalArgumentException(String.format("Field %s does not exist in elements of input array", field));
+        } else {
+          value = value.get(field);
+        }
+      }
+
+      if (value.isObject() ) {
```

Comment:
Should we disallow arrays as well? We could also disallow null nodes, though I'm not sure if it's possible for that to make it here.

Reply:
Hmm, I think disallowing array nodes makes sense here, but I'm not sure we'd want to disallow null nodes, since that's what an optional field would be mapped to.  We could do something like `!value.isValueNode()` which would disallow object, array and missing nodes.